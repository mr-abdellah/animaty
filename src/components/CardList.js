/* eslint-disable react/prop-types */
import '../styles/card.css';
import { useSelector } from 'react-redux';
import Card from './Card';

const CardList = ({ animes }) => {
  const { pending } = useSelector((state) => state.animes);

  return (
    <div className="cards-container">
      {pending ? (
        <h2>Oups Check Your Network ...</h2>
      ) : (
        animes.map((anime) => (
          <Card
            pending={pending}
            key={anime.id}
            title={anime.title}
            description={anime.description}
            id={anime.id}
            img={anime.img}
            type={anime.type}
            episodes={anime.episodes}
            rating={anime.rating}
            year={anime.year}
          />
        ))
      )}
    </div>
  );
};

export default CardList;
