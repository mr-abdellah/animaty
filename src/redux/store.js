import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import animeReducer from './cards';

const store = configureStore({
  reducer: {
    animes: animeReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
