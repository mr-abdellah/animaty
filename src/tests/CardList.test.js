import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import CardList from '../components/CardList';

it(' CardList Test "renders correctly"', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <CardList />
      </Provider>,
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
