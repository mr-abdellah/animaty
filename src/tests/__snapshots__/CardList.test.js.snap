// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` CardList Test "renders correctly" 1`] = `
<div
  className="cards-container"
/>
`;
